cirq/ops/parallel_gate_operation_test.py

```diff
@@ -0,0 +1,176 @@
+    assert cirq.decompose(op) == cirq.X.on_each(qreg)
+
+
+def test_extrapolate():
+    q = cirq.NamedQubit('q')
+    # If the gate isn't extrapolatable, you get a type error.
+    op0 = cirq.ParallelGateOperation(cirq.SingleQubitGate(), [q])
+    with pytest.raises(TypeError):
+        _ = op0**0.5
+
+    op = cirq.ParallelGateOperation(cirq.Y, [q])
+    assert op**0.5 == cirq.ParallelGateOperation(cirq.Y**0.5, [q])
+    assert cirq.inverse(op) == op**-1 == cirq.ParallelGateOperation(cirq.Y**-1,
+                                                                    [q])
+
+
+def test_parameterizable_effect():
+    q = cirq.NamedQubit('q')
+    r = cirq.ParamResolver({'a': 0.5})
+
+    op1 = cirq.ParallelGateOperation(cirq.Z**cirq.Symbol('a'), [q])
+    assert cirq.is_parameterized(op1)
+    op2 = cirq.resolve_parameters(op1, r)
+    assert not cirq.is_parameterized(op2)
+
+
+def test_unitary():
+    a = cirq.NamedQubit('a')
+    b = cirq.NamedQubit('b')
+    g = cirq.SingleQubitGate()
+    p = cirq.ParallelGateOperation(g, [a,b])
+    q = cirq.ParallelGateOperation(cirq.X, [a,b])
+
+    assert not cirq.has_unitary(p)
+    assert cirq.unitary(p, None) is None
+    np.testing.assert_allclose(cirq.unitary(q),
+                               np.array([
+                                   [0. + 0.j, 0. + 0.j, 0. + 0.j, 0. + 0.j],
+                                   [0. + 0.j, 1. + 0.j, 1. + 0.j, 0. + 0.j],
+                                   [0. + 0.j, 1. + 0.j, 1. + 0.j, 0. + 0.j],
+                                   [0. + 0.j, 0. + 0.j, 0. + 0.j, 0. + 0.j]
+                               ]),
+                               atol=1e-8)
+
+
+def test_not_implemented_diagram():
+    a = cirq.NamedQubit('a')
+    g = cirq.SingleQubitGate()
+    c = cirq.Circuit()
+    c.append(cirq.ParallelGateOperation(g,[a]))
+    assert 'cirq.ops.gate_features.SingleQubitGate' in str(c)
+
+
+def test_repr():
+    a, b = cirq.LineQubit.range(2)
+    assert repr(cirq.ParallelGateOperation(cirq.X, (a, b))
+                ) == 'cirq.ParallelGateOperation(gate=cirq.X,' \
+                     ' qubits=[cirq.LineQubit(0), cirq.LineQubit(1)])'
+
+
+def test_str():
+    a, b = cirq.LineQubit.range(2)
+    assert str(cirq.ParallelGateOperation(cirq.X, (a,b))) == 'X(0, 1)'
+
+
+def test_phase():
+    g1 = cirq.SingleQubitGate()
+    g2 = cirq.S
+    g3 = cirq.phase_by(g2, 1, 0)
+    qreg = cirq.LineQubit.range(2)
+    op1 = cirq.ParallelGateOperation(g1, qreg)
+    op2 = cirq.ParallelGateOperation(g2, qreg)
+    assert cirq.phase_by(op2, 1, 0) == cirq.ParallelGateOperation(g3, qreg)
+    with pytest.raises(TypeError):
+        cirq.phase_by(op1, 1, 0)
+
+
+def test_equivalent_circuit():
+    depth = 6
+    qreg = cirq.LineQubit.range(4)
+    oldc = cirq.Circuit()
+    newc = cirq.Circuit()
+    exponents = [0, -1, 1, 1/2, 1/3, 1/3+0.01]
+
+    for moment in range(depth):
+        if moment == 0:
+            gate = cirq.XPowGate(exponent=exponents[moment])
```

Comment:
Make a list of gates and `zip` it with the exponents, instead of making a big if-else chain to manually index into an implicit list.

Reply:
Most of that test was designed to handle coverage with the copied `_formatted_exponent` method. I've cleaned it up with fewer gates but it still effectively performs the same tests.